test/unit/test_parser.py

```diff
@@ -682,6 +720,84 @@ def test__parse_versioned_model_tests(self):
             ["model.snowplow.my_model.v1", "model.snowplow.my_model.v2"],
         )
 
+    def test__parsed_versioned_models(self):
+        block = self.file_block_for(MULTIPLE_TABLE_VERSIONED_MODEL, "test_one.yml")
+        self.parser.manifest.files[block.file.file_id] = block.file
+        dct = yaml_from_file(block.file)
+        self.parser.parse_file(block, dct)
+        self.assert_has_manifest_lengths(self.parser.manifest, nodes=2)
+
+        all_nodes = sorted(self.parser.manifest.nodes.values(), key=lambda n: n.unique_id)
+        models = [node for node in all_nodes if node.resource_type == NodeType.Model]
+
+        # test v1 model
+        parsed_node_patch_v1 = models[0].patch.call_args_list[0][0][0]
```

Comment:
Unfortunately not possible to test on the model nodes directly (e.g. `models[0]`) because these are MagicMock objects and `parser.parse_file` calls `node.patch` under the hood, which mocks the actual patching functionality. @gshank - We'd [brought up moving the `node.patch` method to the PatchParser](https://github.com/dbt-labs/dbt-core/pull/7287#discussion_r1162051159) itself which would make the parsers more testable too - I'm working on opening a tech debt issue for it.

Reply:
Yes, another good reason to do it.